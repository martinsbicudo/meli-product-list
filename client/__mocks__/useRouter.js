jest.mock('next/router', () => ({
  useRouter: () => ({
    push: jest.fn(),
  }),
}));
