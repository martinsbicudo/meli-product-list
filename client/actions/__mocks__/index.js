jest.mock('@Actions', () => ({
  searchProducts: jest.fn().mockImplematation(({
    resolve,
    reject,
    finish,
  }) => {
    resolve();
    reject();
    finish();
  }),
}));
