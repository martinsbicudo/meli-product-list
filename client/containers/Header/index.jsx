import React from 'react';
import { string } from 'prop-types';
import Container from '@Components/Container';

const Header = ({
  search,
}) => (
  <header>
    <Container>
      {search}
    </Container>
  </header>
);

Header.propTypes = {
  search: string,
};

Header.defaultProps = {
  search: '',
};

export default Header;
