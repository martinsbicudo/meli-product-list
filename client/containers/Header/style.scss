.Header {
  background-color: map-get($colors, "meli-turbo");
  padding: 1.5rem 0;

  &Container {
    @include query-max("mobile") {
      flex-wrap: wrap;
      justify-content: center;
    }
  }

  &LogoDesktop,
  &LogoMobile {
    cursor: pointer;

    @include query-max("mobile") { margin-bottom: 2rem; }

    @include query-min("mobile") { margin-right: 2rem; }
  }

  &LogoDesktop {
    @include query-max("mobile") { display: none; }
  }

  &LogoMobile {
    width: 15rem;
    object-fit: contain;

    @include query-min("mobile") { display: none; }
  }

  &Form {
    flex: 1;
    display: flex;

    @include query-max("mobile") { min-width: 100%; }
  }

  &Search {
    box-shadow: 0 0 .2rem map-get($colors, "meli-dusty-gray");

    &Focus {
      transition: .2s box-shadow ease;
      box-shadow: 0 0 1.5rem map-get($colors, "meli-dusty-gray");
    }

    &Image {
      width: 1.6rem;
      height: 1.6rem;
    }
  }

  &SearchImage {
    @include query-max("mobile") { margin-right: 1rem; }
  }

  &SearchSpan {
    @include query-min("mobile") { display: none; }
  }
}
