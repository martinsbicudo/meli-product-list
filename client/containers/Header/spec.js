import '@Client/__mocks__/useRouter'; // eslint-disable-line
import Header from '.'; // eslint-disable-line

describe('Containers - <Header />', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((value) => [value, setState]);
  const createWrapper = () => render(
    <Header />,
  );

  test('Should finds button disabled', () => {
    const { container } = createWrapper();

    expect(container.querySelector('button[type="submit"]')).toBeDisabled();
  });

  test('Should finds button active', () => {
    const { getByPlaceholderText, container } = createWrapper();

    fireEvent.change(getByPlaceholderText('Buscar produto'), {
      target: {
        value: 'TEST_VALUE',
      },
    });

    expect(container.querySelector('button[type="submit"]')).not.toBeDisabled();
  });

  test('Should finds button disabled after submit', () => {
    const { getByPlaceholderText, container } = createWrapper();
    const form = container.querySelector('form');

    fireEvent.change(getByPlaceholderText('Buscar produto'), {
      target: {
        value: 'TEST_VALUE',
      },
    });
    fireEvent.submit(form);

    expect(container.querySelector('button[type="submit"]')).toBeDisabled();
  });

  test('Simulatw focus', () => {
    const { getByPlaceholderText } = createWrapper();

    fireEvent.focus(getByPlaceholderText('Buscar produto'));
  });

  test('Wrapper Snapshot', () => {
    const { asFragment } = createWrapper();

    expect(asFragment()).toMatchSnapshot();
  });
});
