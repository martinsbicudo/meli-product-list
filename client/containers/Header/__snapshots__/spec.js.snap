// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers - <Header /> Wrapper Snapshot 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <div
      class="Container HeaderContainer"
    >
      <img
        alt="Mercado Livre"
        class="HeaderLogoDesktop"
        src="[object Object]"
        srcset="[object Object] 2x"
        title="Mercado Livre"
      />
      <img
        alt="Mercado Livre"
        class="HeaderLogoMobile"
        src="[object Object]"
        srcset="[object Object] 2x"
        title="Mercado Livre"
      />
      <form
        class="HeaderForm"
        method="get"
      >
        <div
          class="ButtonField HeaderSearch"
        >
          <div
            class="FormField ButtonFieldFormField"
          >
            <input
              class="FormFieldInput"
              name="search"
              placeholder="Buscar produto"
              type="text"
              value=""
            />
          </div>
          <button
            class="Button ButtonFieldButton"
            disabled=""
            type="submit"
          >
            <img
              alt="Buscar"
              class="HeaderSearchImage"
              src="[object Object]"
              srcset="[object Object] 2x"
              title="Buscar"
            />
            <span
              class="HeaderSearchSpan"
            >
              Buscar
            </span>
          </button>
        </div>
      </form>
    </div>
  </header>
</DocumentFragment>
`;
